kinto2xml/importer.py

```diff
@@ -0,0 +1,142 @@
+import codecs
+import json
+import requests
+import six
+
+from kinto_client import cli_utils
+from six.moves.urllib.parse import urljoin
+
+from . import constants
+from .amo import prepare_amo_records
+from .kinto import get_kinto_records
+from .logger import logger
+from .synchronize import get_diff, push_changes
+
+
+FIELDS = {
+    'add-ons': ['blockID', 'os', 'guid', 'prefs', 'versionRange', 'details'],
+    'certificates': ['serialNumber', 'issuerName', 'details'],
+    'gfx': ['blockID', 'os', 'vendor', 'feature', 'featureStatus',
+            'driverVersion', 'driverVersionmax', 'driverVersionComparator',
+            'devices', 'details'],
+    'plugins': ['blockID', 'os', 'xpcomabi', 'infoURL', 'versionRange',
+                'matchName', 'matchFilename', 'matchDescription', 'details']
+}
+
+
+def sync_records(amo_records, fields,
+                 kinto_client, bucket, collection, schema, permissions):
+
+    amo_records = prepare_amo_records(amo_records, fields)
+    kinto_records = get_kinto_records(
+        kinto_client=kinto_client,
+        bucket=bucket,
+        collection=collection,
+        schema=schema,
+        permissions=permissions)
+
+    to_create, to_delete = get_diff(amo_records, kinto_records)
+
+    push_changes((to_create, to_delete), kinto_client,
+                 bucket=bucket, collection=collection)
+
+
+def main(args=None):
+    parser = cli_utils.add_parser_options(
+        description='Import the blocklists for AMO into Kinto.',
+        default_collection=None,
+        default_bucket=None,
+        default_server=constants.KINTO_SERVER,
+        default_auth=constants.AUTH,
+        include_bucket=False,
+        include_collection=False)
+
+    parser.add_argument('-S', '--schema-file', help='JSON Schemas file',
+                        type=str, default=constants.SCHEMA_FILE)
+
+    parser.add_argument('--no-schema', help='Should we handle schemas',
```

Comment:
This option is useful in case we start to see that the schema doesn't handle the data we are importing.

Schema is a experimental feature in kinto so we can also have cases where we are importing on a kinto server with deactivated schema.